browse-next/src/components/DeviceSetupDefineMask.vue

```diff
@@ -60,21 +60,28 @@ const device = computed<ApiDeviceResponse | null>(() => {
 });
 
 const getExistingMaskRegions = async () => {
+  let regionsLength = 0;
+  const mostRecentTime = new Date(new Date().setDate(new Date().getDate() + 100));
   if (device.value) {
-    const existingMaskRegions = await getMaskRegionsForDevice(device.value.id);
-    console.log("Lenbth", existingMaskRegions.result.maskRegions.length);
-    for (let i = 0; i < existingMaskRegions.result.maskRegions.length; i++) {
-      points.value = existingMaskRegions.result.maskRegions[i].points;
+    const existingMaskRegions = await getMaskRegionsForDevice(
+      device.value.id,
+      mostRecentTime
+    );
+    // calculate number of regions present
+    while (existingMaskRegions.result[regionsLength]) {
```

Comment:
Why do we need to calculate this?  Can't we just use the array length?

Reply:
I tried that initially, however would always be undefined. I wasn't sure why it was undefined, but tried for a while and couldn't get it to work so have used that solution. If you can get it to work I would prefer to use .length

Reply:
What is the actual structure of the object returned by the api function `getMaskRegionsForDevice`?  It's typed as `Promise<FetchResult<MaskRegionsData>>` which means you should first check that `success` is true, to know what the payload type should be.